de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetBetaWarning.java

```diff
@@ -0,0 +1,237 @@
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.forms.widgets.FormToolkit;
+
+import de.dlr.sc.virsat.external.lib.commons.cli.Activator;
+import de.dlr.sc.virsat.model.dvlm.categories.CategoryAssignment;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.uiengine.ui.editor.snippets.AUiSectionSnippet;
+import de.dlr.sc.virsat.uiengine.ui.editor.snippets.IUiSnippet;
+
+/**
+ *
+ * A snippet that warns if the current element is a BETA element or contains one
+ *
+ */
+public class UiSnippetBetaWarning extends AUiSectionSnippet implements IUiSnippet {
+
+	private static final String SECTION_HEADING = "Beta Warning!";
+	private static final int UI_LAYOUT_NR_COLUMNS = 1;
+	private static final String DEMO_ACTIVE = "demo";
+
+	private final boolean isDemoActive;
+
+	/**
+	 * Constructor for the beta warning snippet
+	 */
+	public UiSnippetBetaWarning() {
+		super();
+		isDemoActive = Activator.getCommandLineManager().isCommandLineOptionSet(DEMO_ACTIVE);
+	}
+	
+	@Override
+	public void createSwt(FormToolkit toolkit, EditingDomain editingDomain, Composite composite, EObject initModel) {
+		super.createSwt(toolkit, editingDomain, composite, initModel);
+
+		Composite sectionBody = createSectionBody(toolkit, SECTION_HEADING, null, UI_LAYOUT_NR_COLUMNS);
+
+		Label label = toolkit.createLabel(sectionBody, createDescriptionLabel());
+
+		Display display = Display.getCurrent();
+		Color red = display.getSystemColor(SWT.COLOR_DARK_RED);
+		Color gray = display.getSystemColor(SWT.COLOR_GRAY);
+		sectionBody.setBackground(red);
+		label.setBackground(red);
+		label.setForeground(gray);
+		
+		setUpLabel(label);
+
+	}
+
+	/**
+	 * Method to set up the label 
+	 * @param label The label to be set up
+	 */
+	private void setUpLabel(Label label) {
+		GridData gridData = createDefaultGridData();
+	    gridData.horizontalSpan = 1;
+		label.setLayoutData(gridData);
+	}
+	
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see
+	 * de.dlr.sc.virsat.uiengine.ui.editor.snippets.IUiSnippet#isActive(org.eclipse.
+	 * emf.ecore.EObject)
+	 */
+	@Override
+	public boolean isActive(EObject model) {
+		if (isDemoActive) {
+			return false;
+		}
+		if (model instanceof StructuralElementInstance) {
+			StructuralElementInstance sei = (StructuralElementInstance) model;
+			for (CategoryAssignment ca : sei.getCategoryAssignments()) {
+				if (isBetaElement(ca.getType())) {
+					return true;
+				}
+			}
+			return isBetaElement(sei.getType());
+		} else if (model instanceof CategoryAssignment) {
+			return isBetaElement(((CategoryAssignment) model).getType());
+		}
+		return false;
+	}
+
+	/**
+	 * Checks if the type definition's containing concept version number starts with
+	 * 0
+	 * 
+	 * @param typeDefinition
+	 *            the type definition
+	 * @return if it is a beta element or not
+	 */
+	protected boolean isBetaElement(EObject typeDefinition) {

```

Comment:
Let's maybe decouple beta check from the version number? Let's say my concept is v3, but it's still considered a beta (like Scosa)
We could add a beta keyword in the concept grammar so that you explicitly matk your concept version as beta